### PR TITLE
Remove the use of six and __future__ 

### PR DESCRIPTION
**Pull request by sechkova**
Fixes: #353 
Fixes: #354

### Description of the changes being introduced by the pull request:

Since the project no longer supports Python2, imports of `six` and `__future__` are no longer needed.

### Please verify and check that the pull request fulfils the following requirements:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: tests/test_hash.py [hunk 1]
```diff
@@ -259,7 +250,7 @@ def test_update_file_obj(self):
 
   def _do_update_file_obj(self, library, algorithm):
     data = 'abcdefgh' * 4096
-    file_obj = six.StringIO()
+    file_obj = io.StringIO()
```
Comment (jku) [inline tests/test_hash.py:253]: I did not even know this existed :eyes: ... seems nice

Comment (sechkova) [reply tests/test_hash.py:253]: same here

**Comment by sechkova** [comment]
I double checked that I removed all direct 'six' dependencies in the project but can someone with a better understanding confirm whether 'six' still being installed during the CI run is expected (a dependency of some dependency? still dependency of the released sslib version?) .

From the (3.8, ubuntu-latest, py) build:
```
Collecting tox
  Using cached tox-3.23.1-py2.py3-none-any.whl (85 kB)
Collecting six>=1.14.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
```

```
Run tox -e py
py create: /home/runner/work/securesystemslib/securesystemslib/.tox/py
py installdeps: -r/home/runner/work/securesystemslib/securesystemslib/requirements-pinned.txt, -r/home/runner/work/securesystemslib/securesystemslib/requirements-test.txt
py installed: cffi==1.14.5,colorama==0.4.4,coverage==5.6b1,cryptography==3.4.7,pycparser==2.20,PyNaCl==1.4.0,six==1.16.0
```

**Comment by jku** [comment]
```
Collecting six>=1.14.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
```
I think this output is from the actual container install (where we just pip install tox) -- tox somehow depending on six is likely.

```
py installed: cffi==1.14.5,colorama==0.4.4,coverage==5.6b1,cryptography==3.4.7,pycparser==2.20,PyNaCl==1.4.0,six==1.16.0
```
This is from inside the tox environment so more interesting: pynacl still seems to pull in six (although next release should stop that)

**Review by jku — APPROVED** [review]
LGTM

File: tests/test_gpg.py [hunk 2]
```diff
@@ -30,7 +30,6 @@
 else:
   from mock import patch # pylint: disable=import-error
```
Comment (joshuagl) [inline tests/test_gpg.py:31]: Hmm, can we also drop our mock dependency?

**Review by joshuagl — APPROVED** [review]
LGTM, thanks @sechkova !

**Pull request merged**